lab_4_fill_words_by_ngrams/main.py

```diff
@@ -273,6 +424,16 @@ def _load_from_json(self) -> dict[tuple[str, int], str]:  # type: ignore
                 or if attribute _json_path has inappropriate extension,
                 or if inappropriate type loaded data.
         """
+        if not isinstance(self._json_path, str):
+            raise ValueError('Inappropriate type of attribute _json_path')
+        if not self._json_path:
+            raise ValueError('Attribute _json_path is empty')
+        if self._json_path[-4:] != 'json':
+            raise ValueError('Attribute _json_path has inappropriate extension')
+        with open(self._json_path, 'r', encoding="utf-8") as file:
+            self._questions_and_answers = json.load(file)
```

Comment:
So the formatting does not need to be changed?

Oh i see, you do it below. Well, consider making this iteration once instead of doing it every time the questions are requested

Reply:
fixed